kong/plugins/opentelemetry/handler.lua

```diff
@@ -23,41 +23,36 @@ local translate_span_trace_id = otlp.translate_span
 local encode_span = otlp.transform_span
 local to_hex = require "resty.string".to_hex
 
+
 local _log_prefix = "[otel] "
 
+
 local OpenTelemetryHandler = {
   VERSION = "0.1.0",
   PRIORITY = 14,
 }
 
-local default_headers = {
-  ["Content-Type"] = "application/x-protobuf",
+local CONTENT_TYPE_HEADER_NAME = "Content-Type"
+local DEFAULT_CONTENT_TYPE_HEADER = "application/x-protobuf"
+local DEFAULT_HEADERS = {
+  [CONTENT_TYPE_HEADER_NAME] = DEFAULT_CONTENT_TYPE_HEADER
 }
 
 -- worker-level spans queue
-local queues = {} -- one queue per unique plugin config
-local headers_cache = setmetatable({}, { __mode = "k" })
+local QUEUES = {} -- one queue per unique plugin config
 
 
-local function get_cached_headers(conf_headers)
-  if not conf_headers then
-    return default_headers
+local function get_headers(conf_headers)
+  if not conf_headers or conf_headers == null then
+    return DEFAULT_HEADERS
   end
 
-  -- cache http headers
-  local headers = headers_cache[conf_headers]
-
-  if not headers then
-    headers = clone(default_headers)
-    if conf_headers and conf_headers ~= null then
-      for k, v in pairs(conf_headers) do
-        headers[k] = v
-      end
-    end
-
-    headers_cache[conf_headers] = headers
+  if conf_headers[CONTENT_TYPE_HEADER_NAME] then
```

Comment:
Not familiar with the code, so please bear with me.
It isn't immediately clear (to me) what the reason is for checking content-type specifically; is it being used a flag? Could we make it clearer perhaps with a comment if that is the intent?
Could there be a case (in the previous revision) where `conf_headers` does not include the content-type -- and now it always does?

Reply:
@gszr, It is checked here to skip modifying table, you can see below it clones conf headers and adds it.